hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -3075,6 +3080,13 @@ private RegionScannerImpl getScanner(Scan scan, List<KeyValueScanner> additional
           checkFamily(family);
         }
       }
+      if((Bytes.equals(scan.getStartRow(), getRegionInfo().getStartKey())
```

Comment:
Can add a comment saying what all we count as full scan..  the check is bit complex,.. So nice code level comment helps..  Also we can do this boolean eval at create of scan itself and use that boolean state? No need to do all thse bytes[] checks for every next() call?

Reply:
done, now it is at time scanner instantiated only.